jbpm-flow-builder/src/main/java/org/jbpm/assembler/AbstractProcessAssembler.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.assembler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.drools.compiler.builder.impl.KnowledgeBuilderImpl;
+import org.drools.compiler.compiler.BaseKnowledgeBuilderResultImpl;
+import org.drools.compiler.compiler.PackageRegistry;
+import org.drools.compiler.compiler.ProcessLoadError;
+import org.drools.core.impl.InternalKnowledgeBase;
+import org.jbpm.compiler.ProcessBuilderImpl;
+import org.kie.api.definition.process.Process;
+import org.kie.api.internal.assembler.KieAssemblerService;
+import org.kie.api.io.Resource;
+import org.kie.api.io.ResourceConfiguration;
+import org.kie.api.io.ResourceType;
+import org.kie.internal.builder.KnowledgeBuilderResult;
+
+public abstract class AbstractProcessAssembler implements KieAssemblerService {
```

Comment:
Take this as nitpicking it's ok :)
kb, kbase,e,es, pkg
doesnt intellij have like auto-suggestions for naming fields based on their type? 

Real question - why are we adding all exceptions except RuntimeException which is being re-thrown? Just curious.

Reply:
probably an oversight; the `addResource()` signature includes `throws Exception`, and `e` here have not been casted. Will fix. (In fact: notice this originally was a quasi-verbatim copy of [KnowledgeBuilderImpl#addProcessFromXml()](https://github.com/kiegroup/drools/blob/8092938bc75726481deb26e94c70d8167c7ba6e9/drools-compiler/src/main/java/org/drools/compiler/builder/impl/KnowledgeBuilderImpl.java#L703-L737) where the cast exists)